### PR TITLE
Completion: fix qualified completion in sequence expressions

### LINKED ISSUE #18033: Member completion uses wrong qualifier
Consider the following example:

```fsharp
let f (s: string) =
    s.Trim()
    s.Trim()
    s.Trim()
    ()
```

It's expected that the member code completion is the same after each `s.Trim()` invocation, however, members are suggested in the first case only, and the next two cases suggest unqualified items:

Image
Image

Initially reported in [RIDER-120178](https://youtrack.jetbrains.com/issue/RIDER-120178).

### PR DESCRIPTION
**Pull request by auduchinok**
Fixes #18033. I'm not entirely sure what the removed lines were intended to do, so maybe removing the reverts the fix introduced in #16882. The test added in #16882 passes for locally with this change.

### CODE REVIEW

File: src/Compiler/Service/ServiceParseTreeWalk.fs [hunk 1]
```diff
@@ -388,12 +388,6 @@ module SyntaxTraversal =
                 seq {
                     match expr with
                     | SynExpr.Sequential(expr1 = expr1; expr2 = SynExpr.Sequential _ as expr2) ->
-                        // It's a nested sequential expression.
-                        // Visit it, but make defaultTraverse do nothing,
-                        // since we're going to traverse its descendants ourselves.
-                        yield dive expr expr.Range (fun expr -> visitor.VisitExpr(path, traverseSynExpr path, (fun _ -> None), expr))
```
Comment (T-Gro) [inline src/Compiler/Service/ServiceParseTreeWalk.fs:394]: expr1 is "dive"'d right below this call, and expr2 is handled via the recursive invocation => cannot see any impact from ommiting dive into the expr itself.

=> LGTM.

**Review by T-Gro — APPROVED** [review]

File: src/Compiler/Service/ServiceParseTreeWalk.fs (hunk 1 above)
Comment (brianrourkeboll) [reply src/Compiler/Service/ServiceParseTreeWalk.fs:394]: Does `visitor.VisitExpr` still get called on `expr` itself now though? Maybe it does and I'm just not awake yet :)

**Comment by auduchinok** [comment]
OK, everything, except the release notes, is green.

**Comment by auduchinok** [comment]
This has been green for a while, but the release notes constantly need resolving conflicts.

**Comment by psfinaki** [comment]
Yeah release notes conflicts can be annoying, tho we sometimes resolve them to keep PRs in shape :)

This LGTM, I was just also wondering about what Brian mentioned above.

**Review by psfinaki — APPROVED** [review]

**Comment by auduchinok** [comment]
Can this be merged?

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**